File: ncarg2d/src/libncarg/tdpack/tdstrs.f
C
C $Id: tdstrs.f,v 1.5 2008-07-27 00:17:33 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE TDSTRS (IRST,IA01,IA02,IA03,IA04,IA05,IA06,IA07,RA08,
     +                                                      RA09,RA10)
C
C This routine allows the caller to set the values defining a particular
C rendering style.  The arguments are as follows (all are input, those
C beginning with an "I" are of type INTEGER, those beginning with an "R"
C are of type "REAL"):
C
C   IRST is the index of the rendering style for which values are to
C   be set and should be in the range from 1 to 64, inclusive.
C
C   IA01 and IA02 are color indices specifying a range of colors to be
C   used for the "bottom" side of a surface (where function values are
C   less than the value on the surface).  If IA01 is negative, filling
C   of triangles seen from the "bottom" is turned off.  If IA01 is zero
C   or greater, but IA02 is less than or equal to it, the color with
C   index IA01 is used.  If IA01 is zero or greater and IA02 is greater
C   than IA01, then a range of color indices is specified; colors near
C   the beginning of that range are used for triangles that are nearly
C   perpendicular to the line of sight, while colors near the end of
C   that range are used for triangles more nearly parallel to the line
C   of sight.
C
C   IA03 and IA04 are color indices specifying a range of colors to be
C   used for the "top" side of a surface (with function values greater
C   than the value in the surface).  If IA03 is negative, filling
C   of triangles seen from the "top" is turned off.  If IA03 is zero
C   or greater, but IA04 is less than or equal to it, the color with
C   index IA03 is used.  If IA03 is zero or greater and IA04 is greater
C   than IA03, then a range of color indices is specified; colors near
C   the beginning of that range are used for triangles that are nearly
C   perpendicular to the line of sight, while colors near the end of
C   that range are used for triangles more nearly parallel to the line
C   of sight.
C
C   IA05 is the color index specifying a color to be used for lines
C   drawn on the "bottom" side of a surface.  If IA05 is negative, the
C   drawing of these lines is turned off.
C
C   IA06 is the color index specifying a color to be used for lines
C   drawn on the "top" side of a surface.  If IA06 is negative, the
C   drawing of these lines is turned off.
C
C   IA07 is a flag, which, if set non-zero, turns on the drawing of
C   the boundaries of the individual triangles into which surfaces
C   have been decomposed.
C
C   RA08, RA09, and RA10 are the distances between slices in the U, V,
C   and W directions, respectively.
C
C Declare a necessary TDPACK common block.
C
        COMMON /TDCOM3/ IFC1(64),IFC2(64),IFC3(64),IFC4(64),ILC1(64)
        COMMON /TDCOM3/ ILC2(64),ILTD(64),USTP(64),VSTP(64),WSTP(64)
        SAVE   /TDCOM3/
C
C Copy the values into common.
C
        INDX=MAX(1,MIN(64,IRST))
        IFC1(INDX)=IA01
        IFC2(INDX)=IA02
        IFC3(INDX)=IA03
        IFC4(INDX)=IA04
        ILC1(INDX)=IA05
        ILC2(INDX)=IA06
        ILTD(INDX)=IA07
        USTP(INDX)=RA08
        VSTP(INDX)=RA09
        WSTP(INDX)=RA10
C
C Done.
C
        RETURN
C
      END

File: ncarg2d/man/tdpack/tdstrs.m
.TH TDSTRS 3NCARG "July 1997" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
TDSTRS - Set the values defining a selected rendering style, which affects the
appearance of triangles drawn by a call to the routine TDDTRI.
.SH SYNOPSIS
CALL TDSTRS (IRST, IFC1, IFC2, IFC3, IFC4, ILC1, ILC2, ILTD, USTP, VSTP, WSTP)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_tdstrs(int irst, int ifc1, int ifc2, int ifc3, int ifc4, int ilc1,
int ilc2, int iltd, float ustp, float vstp, float wstp)
.SH DESCRIPTION
The arguments of TDSTRS are as follows:
.IP "IRST" 8
(an input expression of type INTEGER) - the index of the particular rendering
style which is to be retrieved.
.IP "IFC1 and IFC2" 8
(output variables of type INTEGER) -
color indices specifying a range of colors to be used
for the "bottom" side of a surface (where function values are less than
the value on the surface). If IFC1 is negative, filling of triangles seen
from the "bottom" is turned off. If IFC1 is zero or greater, but IFC2 is
less than or equal to it, the color with index IFC1 is used. If IFC1 is zero
or greater and IFC2 is greater than IFC1, then a range of color indices
is specified; colors near the beginning of that range are used for
triangles that are nearly perpendicular to the line of sight, while colors
near the end of that range are used for triangles more nearly parallel
to the line of sight. (Normally, one should make triangles
perpendicular to the line of sight lighter than those parallel to the line
of sight.)
.IP "IFC3 and IFC4" 8
(output variables of type INTEGER) -
color indices specifying a range of colors to be used
for the "top" side of a surface (where function values are greater than
the value on the surface). If IFC3 is negative, filling of triangles seen
from the "top" is turned off. If IFC3 is zero or greater, but IFC4 is less
than or equal to it, the color with index IFC3 is used. If IFC3 is zero or
greater and IFC4 is greater than IFC3, then a range of color indices is
specified; colors near the beginning of that range are used for triangles
that are nearly perpendicular to the line of sight, while colors near the
end of that range are used for triangles more nearly parallel to the line
of sight. (Normally, one should make triangles perpendicular to the
line of sight lighter than those parallel to the line of sight.)
.IP "ILC1" 8
(an output variable of type INTEGER) -
the color index specifying a color to be used for lines drawn on
the "bottom" side of a surface.  If ILC1 is negative, the drawing of these
lines is turned off.
.IP "ILC2" 8
(an output variable of type INTEGER) -
the color index specifying a color to be used for lines drawn on
the "top" side of a surface.  If ILC2 is negative, the drawing of these lines
is turned off.
.IP "ILTD" 8
(an output variable of type INTEGER) -
a flag, which, if set non-zero, turns on the drawing of the
edges of the individual triangles into which surfaces have been
decomposed.
.IP "USTP, VSTP, and WSTP" 8
(output variables of type REAL) - the distances between slices in the U, V,
and W directions, respectively. If a given value is zero, the associated
slice lines are not drawn.
.SH C-BINDING DESCRIPTION 
The C-binding argument descriptions are the same as the FORTRAN 
argument descriptions.
.SH ACCESS
To use TDSTRS or c_tdstrs, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order. 
.SH SEE ALSO
Online:
tdclrs, tdctri, tddtri, tdgeti, tdgetr, tdgrds, tdgrid, tdgtrs, tdinit, tditri,
tdlbla, tdlbls, tdline, tdlnpa, tdmtri, tdotri, tdpack, tdpack_params,
tdpara, tdplch, tdprpa, tdprpi, tdprpt, tdseti, tdsetr, tdsort, tdstri
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
